core/model/src/main/java/org/eclipse/sw360/antenna/model/artifact/facts/ArtifactCoordinates.java

```diff
@@ -11,10 +11,36 @@
 
 package org.eclipse.sw360.antenna.model.artifact.facts;
 
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
 public abstract class ArtifactCoordinates<T extends ArtifactCoordinates> implements ArtifactIdentifier<T> {
     public abstract String getName();
+
     public abstract String getVersion();
 
+    public abstract String getType();
+
+    public String getPurl() {
+        return String.format("pkg:%s/%s%s", percentEncoding(getType()), percentEncoding(getName()),
```

Comment:
The `type` must not be percent-encoded, see https://github.com/package-url/purl-spec#rules-for-each-purl-component.